File: C#/Day12/HotelBookingSystem/Models/Room.cs
namespace HotelBookingSystem.Models
{


    public enum RoomStatus{
        Available,
        Booked,
        UnderMaintaenance
    }

  

    public class Room
    {
        public int Id { get; set; }

        public string RoomNumber { get; set; } = null!;

        public RoomStatus Status { get; set; }
        public decimal PricePerNight { get; set; }
        public int RoomTypeId { get; set; }

        public int HotelId { get; set; }

        public Hotel Hotel { get; set; } = null!;

        public RoomType RoomType { get; set; } = null!;

        public List<Booking> Bookings { get; set; } = new List<Booking>();

    }   
}

File: Angular/Hotel_Booking_System/src/app/shared/models/room.ts
export enum RoomStatus {
    AVAILABLE = 'Available',
    BOOKED = 'Booked',
    MAINTENANCE = 'Maintenance'
}


export interface RoomType{
  id: string;
  typeName: string;
  description: string;
  capacity: number;
}
export interface Room {
    id: string;
    hotelId: string;
    roomNumber: string;
    floor: number;
    pricePerNight: number;
    roomType: RoomType;
    status: RoomStatus;
}
